aws/resource_aws_worklink_website_certificate_authority_association.go

```diff
@@ -0,0 +1,170 @@
+}
+
+func resourceAwsWorkLinkWebsiteCertificateAuthorityAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).worklinkconn
+
+	websiteCaID, fleetArn, err := decodeWorkLinkWebsiteCertificateAuthorityAssociationResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	input := &worklink.DescribeWebsiteCertificateAuthorityInput{
+		FleetArn:    aws.String(fleetArn),
+		WebsiteCaId: aws.String(websiteCaID),
+	}
+
+	resp, err := conn.DescribeWebsiteCertificateAuthority(input)
+	if err != nil {
+		if isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
+			log.Printf("[WARN] WorkLink Website Certificate Authority Association (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error describing WorkLink Website Certificate Authority Association (%s): %s", d.Id(), err)
+	}
+
+	d.Set("website_ca_id", websiteCaID)
+	d.Set("fleet_arn", fleetArn)
+	d.Set("certificate", resp.Certificate)
+	d.Set("display_name", resp.DisplayName)
+
+	return nil
+}
+
+func resourceAwsWorkLinkWebsiteCertificateAuthorityAssociationDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).worklinkconn
+
+	websiteCaID, fleetArn, err := decodeWorkLinkWebsiteCertificateAuthorityAssociationResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	input := &worklink.DisassociateWebsiteCertificateAuthorityInput{
+		FleetArn:    aws.String(fleetArn),
+		WebsiteCaId: aws.String(websiteCaID),
+	}
+
+	if _, err := conn.DisassociateWebsiteCertificateAuthority(input); err != nil {
+		if isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
+			return nil
+		}
+		return fmt.Errorf("Error deleting WorkLink Website Certificate Authority Association (%s): %s", d.Id(), err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"DELETING"},
+		Target:     []string{"DELETED"},
+		Refresh:    worklinkWebsiteCertificateAuthorityAssociationStateRefresh(conn, websiteCaID, fleetArn),
+		Timeout:    15 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for disassociate Worklink Website Certificate Authority (%s) to become deleted: %s",
+			d.Id(), err)
+	}
+
+	return nil
+}
+
+func worklinkWebsiteCertificateAuthorityAssociationStateRefresh(conn *worklink.WorkLink, websiteCaID, arn string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		emptyResp := &worklink.DescribeWebsiteCertificateAuthorityOutput{}
+
+		resp, err := conn.DescribeWebsiteCertificateAuthority(&worklink.DescribeWebsiteCertificateAuthorityInput{
+			FleetArn:    aws.String(arn),
+			WebsiteCaId: aws.String(websiteCaID),
+		})
+		if err != nil {
+			if isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
+				return emptyResp, "DELETED", nil
+			}
+		}
```

Comment:
If the error is not `worklink.ErrCodeResourceNotFoundException`, we should return the error.

```suggestion
			return nil, "", err
		}
```

For what its worth, we have been slowly trying to move away from the convention of using `if isAWSErr()` conditionals inside the `if err != nil` conditionals to make this type of coding mistake more obvious, e.g. in this case

```go
		if isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
			return emptyResp, "DELETED", nil
		}

		if err != nil {
			return nil, "", err
		}
```

Its a bummer this cannot be caught by linting as its a logic problem. 😄

Reply:
Fixed.